R/zchunk_LB164.ag_Costs_USA_C_2005_irr.R

```diff
@@ -36,43 +42,124 @@ module_aglu_LB164.ag_Costs_USA_C_2005_irr_DISABLED <- function(command, ...) {
     USDA_cost_data <- get_data(all_data, "aglu/USDA_cost_data")
     L100.LDS_ag_HA_ha <- get_data(all_data, "L100.LDS_ag_HA_ha")
     L133.ag_Cost_75USDkg_C <- get_data(all_data, "L133.ag_Cost_75USDkg_C")
-    L161.ag_irrHA_frac_R_C_GLU <- get_data(all_data, "L161.ag_irrHA_frac_R_C_GLU")
+    L161.ag_irrHA_frac_R_C_GLU <- get_data(all_data, "temp-data-inject/L161.ag_irrHA_frac_R_C_GLU")
+
+
+    # Perform computations
+    # The method here is to start from the cost file that has already been processed in LB133, and
+    # to deduct costs of purchased irrigation water from each cost estimate.
+
+    # Step 1: Compute the share of total variable costs from purchased irrigation water.
+    # CostFrac = IrrCost / TotCost
+    USDA_cost_data %>%
+      gather(year, cost, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      filter(Item == "Purchased irrigation water" | Item == "Total operating costs",
+             year %in% MODEL_COST_YEARS) %>%
+      spread(Item, cost) %>%
+      rename(IrrCost = `Purchased irrigation water`,
+             TotCost = `Total operating costs`) %>%
+      mutate(CostFrac = IrrCost / TotCost) %>%
+      # filter out any crops with IrrCost in all years
```

Comment:
whoops, that's an incomplete comment. We filter out any crops missing IrrCost in all MODEL_COST_YEARS. So if they have an IrrCost in at least one of the years, they stay